lib/community/pages/community_page.dart

```diff
@@ -122,44 +123,17 @@ class _CommunityPageState extends State<CommunityPage> with AutomaticKeepAliveCl
         },
         listener: (context, state) {
           if (state.status == CommunityStatus.failure) {
-            SnackBar snackBar = SnackBar(
-              content: Text(state.errorMessage ?? AppLocalizations.of(context)!.missingErrorMessage),
-              behavior: SnackBarBehavior.floating,
-            );
-            WidgetsBinding.instance.addPostFrameCallback((timeStamp) => ScaffoldMessenger.of(context).showSnackBar(snackBar));
+            showSnackbar(context, state.errorMessage ?? AppLocalizations.of(context)!.missingErrorMessage);
           }
 
           if (state.status == CommunityStatus.success && state.blockedCommunity != null) {
-            SnackBar snackBar = SnackBar(
-              content: Row(
-                mainAxisAlignment: MainAxisAlignment.spaceBetween,
-                children: [
-                  Expanded(
-                    child: Text(
-                      'Successfully ${state.blockedCommunity?.blocked == true ? 'blocked' : 'unblocked'} ${state.blockedCommunity?.communityView.community.title}',
-                      overflow: TextOverflow.ellipsis,
-                    ),
-                  ),
-                  if (state.blockedCommunity?.blocked == true)
-                    SizedBox(
-                      height: 20,
-                      child: IconButton(
-                        visualDensity: VisualDensity.compact,
-                        onPressed: () {
-                          ScaffoldMessenger.of(context).clearSnackBars();
-                          context.read<CommunityBloc>().add(BlockCommunityEvent(communityId: state.blockedCommunity!.communityView.community.id, block: false));
-                        },
-                        icon: Icon(
-                          Icons.undo_rounded,
-                          color: theme.colorScheme.inversePrimary,
-                        ),
-                      ),
-                    ),
-                ],
-              ),
-              behavior: SnackBarBehavior.floating,
-            );
-            WidgetsBinding.instance.addPostFrameCallback((timeStamp) => ScaffoldMessenger.of(context).showSnackBar(snackBar));
+            if (state.blockedCommunity?.blocked == true) {
+              showSnackbar(context, AppLocalizations.of(context)!.successfullyBlockedCommunity(state.blockedCommunity?.communityView.community.title ?? 'UNKNOWN'), undoAction: () {
```

Comment:
Maybe we can use `missingErrorMessage` localization message for the default here too!